--- a/Common/EncoderDev/HAPIHost/CHapi.cpp.orig	2018-01-10 16:39:51.562419718 -0700
+++ a/Common/EncoderDev/HAPIHost/CHapi.cpp	2018-01-10 16:41:09.441100440 -0700
@@ -1590,7 +1590,7 @@
 	Parameter_Map_Item *p_map = NULL;
 	Uint32 currentType = pParam->ParamType;
 
-#if 1//DBG
+#if 0//DBG
 	char param_dummy[64];
 	param_dummy[0] = '-'; //reduce spew for unset overrides
 	strcpy(&param_dummy[1], param_name);
@@ -1599,6 +1599,14 @@
 		DbgLogInfo(("set_param(%p, %s, %u) IGNORED due to registry setting\n", pParam, param_name, value, currentType));
 		return;
 	}
+#else
+        // Use registered settings if they exist.
+        Uint32 reg_value;
+        if (hcwGetRegistryDWORD(param_name, (LPDWORD)&reg_value))
+        {
+            DbgLogInfo(("set_param(%p, %s, %u) override by registry %u\n", pParam, param_name, value, reg_value));
+            value = reg_value;
+        }
 #endif
 
     //Special case, since we need a valid ParamType to establish which table to use
