--- a/Common/EncoderDev/HAPIHost/Hapi.cpp-bak	2020-05-16 15:56:13.725294522 -0600
+++ b/Common/EncoderDev/HAPIHost/Hapi.cpp	2020-05-16 15:56:29.304064326 -0600
@@ -300,6 +300,7 @@
 
 static void DLLCALL cmdThreadFunc(void * data)
 {
+	setThreadName("HAPI-cmd");
 	HAPIHandler hapi = (HAPIHandler)data;
 	QueueItem item;
 	Boolean toSend ;
@@ -856,6 +857,7 @@
 #ifndef AV_ASI
 static void DLLCALL dataThreadFunc(void * data)
 {
+	setThreadName("HAPI-data");
 	HAPIHandler hapi = (HAPIHandler)data;
 	QueueItem item;
 	Boolean toSend ;
@@ -1211,6 +1213,7 @@
 
 static void DLLCALL mainThreadFunc(void * data)
 {
+	setThreadName("HAPI-main");
 	HAPIHandler hapi = (HAPIHandler)data;
 
 	QueueItem item;
@@ -1513,6 +1516,7 @@
 #ifdef USE_WRITE_THREAD
 static void DLLCALL writeThreadFunc(void * data)
 {
+	setThreadName("HAPI-write");
 	HAPIHandler hapi = (HAPIHandler)data;
 
 	QueueItem item;
@@ -1644,6 +1648,7 @@
  */
 static void DLLCALL callbackThreadFunc(void * data)
 {
+	setThreadName("HAPI-callback");
 	HAPIHandler hapi = (HAPIHandler)data;
 
 	QueueItem item;
